metric_learn/base_metric.py

```diff
@@ -296,6 +296,7 @@ def get_mahalanobis_matrix(self):
 
 class _PairsClassifierMixin(BaseMetricLearner):
 
+  classes_ = [-1, 1]
   _tuple_size = 2  # number of points in a tuple, 2 for pairs
```

Comment:
shouldn't we document the attribute `threshold_` here in the general class rather than in the downstream classes?

Reply:
I guess I should document it for both abstract and downstream classes, but I don't think I can document it only here: indeed, we can inherit docstrings, but I don't think we can inherit docstrings paragraph-wise (e.g. `_PairsClassifierMixin` has an attribute `threshold_` that we want to transmit to `MMC`. But the problem is that `MMC` has also a title, and also other attributes like `transformer_`). I didn't find an option to do this in sphinx and this impossibility make some sense because if we write two docstrings paragraphs (below the title), would the result in the downstream class be the concatenation of both descriptions ? It could give weird results (e.g. : "This is the abstract class", "This is is the MMC class", would give: "This is the abstract class. This is the MMC class") However there seem to exist some packages that could allow to do something like this. We could use them but maybe later on in the development if things get too replicated. https://github.com/Chilipp/docrep https://github.com/meowklaski/custom_inherit
Then there's also the solution not to care about inheriting the docstrings, but I think it's better for a user to know what are all the attributes in his object without needing to click on the doc of all mother classes (that's what is done in scikit-learn for instance I think)
Finally there's also another option which is to declare attributes as properties so that we can have a more granular inheritance, but I think this adds to much complexity just for docstrings

Reply:
(done: I just added them to the base PairsClassifier class too)